engine/src/flutter/shell/platform/embedder/embedder.h

```diff
@@ -2631,6 +2631,11 @@ typedef struct {
   /// the native view. The callback is invoked from a task posted to the
   /// platform thread.
   FlutterViewFocusChangeRequestCallback view_focus_change_request_callback;
+
+  /// Opaque identifier provided by the engine. Accessible in Dart code through
+  /// `PlatformDispatcher.instance.engineId`. Can be used in native code to
+  /// retrieve the engine instance that is running the Dart code.
+  int64_t engine_id;
```

Comment:
@chinmaygarde Do you think we should create a `FlutterEngineId` typedef? Or is that over-kill?

For example, here's the typedef we have for `FlutterViewId`:

https://github.com/flutter/flutter/blob/50a1b953755b752f2a7823e381b41dbb54386cbd/engine/src/flutter/shell/platform/embedder/embedder.h#L277-L281